src/core/defcomp.jl

```diff
@@ -82,6 +82,11 @@ function add_variable(comp_id::ComponentId, name, datatype, dimensions, descript
 end
 
 function add_parameter(comp_def::ComponentDef, name, datatype, dimensions, description, unit, default)
+    if default !== nothing
+        if length(dimensions) != ndims(default)
+            error("Default value has different number of dimensions ($(ndims(dflt))) than parameter '$name' ($(length(dimensions)))")
+        end
```

Comment:
ah actually it is just the "end" line of the if statement that codecov is complaining about. But I was surprised that the error type for that error would be `UndefVarError` and not still a `LoadError`, so I think there might be a problem here that `dflt` isn't defined here and that it should be `default`?

Reply:
Ah, yes, good catch! The test is now reporting a bug in the line that throws the error, i.e. in line 87 :) The test should actually test for an `ErrorException`, right? That is what is thrown, so any other type of error would mean something went wrong.

Reply:
Good point, I was lazy when I changed that, I’ll change the error type back and correct dflt to default, I think that should solve it.

Reply:
Sorry, but did we conclude that it should be a `LoadError` or `ErrorException`?  It's still an `ErrorException` (see ), but even with `default` instead of `dflt`. Is there a problem here because we're calling `ndims` on `default` before it is evaluated ... or has it been evaluated at this point?

Reply:
I think David is right, it should now be `ErrorException` (it was previously `LoadException` when it was being thrown from within eval I think)

Reply:
Also if you just run the part of the test file that's supposed to test this error, but run it outside of the `test_throws` then you can see if the error message is our constructed error message or if something else is wrong, just to be safe

Reply:
@ckingdon95 smart yes I just did that and the constructed error message looks correct, so I think we can merge the PR now if someone can approve it